h5io/_h5io.py

```diff
@@ -193,14 +188,20 @@ def _triage_write(key, value, root, comp_kw, where,
                       where + '.csr_matrix_shape', cleanup_data=cleanup_data,
                       slash=slash)
     else:
-        if isinstance(value, (DataFrame, Series)):
-            if isinstance(value, DataFrame):
-                title = 'pd_dataframe'
-            else:
-                title = 'pd_series'
-            rootname = root.name
-            cleanup_data.append({title: (rootname, key, value)})
+        try:
+            from pandas import DataFrame, Series
+        except ImportError:
+            pass
         else:
+            if isinstance(value, (DataFrame, Series)):
+                if isinstance(value, DataFrame):
+                    title = 'pd_dataframe'
+                else:
+                    title = 'pd_series'
+                rootname = root.name
+                cleanup_data.append({title: (rootname, key, value)})
+                return
+
             err_str = 'unsupported type %s (in %s)' % (type(value), where)
             raise TypeError(err_str)
```

Comment:
Yes, overlooked that on my final edit last night. My local `nosetests`
where passing so I thought it safe to commit. I've fixed this and pushed a
new version. I'm not sure why the coverage decreased so much though.

On Sat, Jun 25, 2016 at 4:45 AM, Eric Larson <notifications@github.com>
wrote:

> In h5io/_h5io.py
> <https://github.com/h5io/h5io/pull/18#discussion_r68487890>:
>
> >          else:
> > +            if isinstance(value, (DataFrame, Series)):
> > +                if isinstance(value, DataFrame):
> > +                    title = 'pd_dataframe'
> > +                else:
> > +                    title = 'pd_series'
> > +                rootname = root.name
> > +                cleanup_data.append({title: (rootname, key, value)})
> > +                return
> > +
> >              err_str = 'unsupported type %s (in %s)' % (type(value), where)
> >              raise TypeError(err_str)
>
> I think these still need to be unindented one level to match my snippet,
> and Travis agrees
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/h5io/h5io/pull/18/files/3f013a70523ca8dba90a1ae1b3f86c212e6c49cc#r68487890>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABXvSA0qrFg0ivHVOxn4O64iM3xQggXqks5qPRTNgaJpZM4I98XJ>
> .
>